cypress/integration/crud.spec.ts

```diff
@@ -0,0 +1,50 @@
+const book_title = 'The Little Schemer';
+
+describe('CRUD Actions', () => {
+	context('Valid Data', function(){
+		beforeEach(() => {
+			cy.visit('/#/examples/crud');
+			cy.get('.add').click();
+			cy.get('#mat-input-0').type(book_title);
+			cy.get('#mat-input-1').type('Friedman{enter}');
+		});
+	  
+		it('should display a book added', () => {
+			cy.get('mat-card>h3').should('contain', book_title);
+		});
+	  
+		it('should not display a book deleted', () => {
+			cy.get('mat-card').first().click();
+			cy.get('[ng-reflect-message="Delete book"]').click();
```

Comment:
> Same remark here.

mat-card is a component and ng-reflect-message attribute. So, here too should I add title for delete action? Is the usage of mat-card inappropriate?

Reply:
I believe `mat-card` is inappropriate too, if you would change `mat-card` with your own component `ngx-card` these tests would fail. Therefor I rarely use these selectors in my tests, but like everything there are always exceptions to this rule.

To fix this we could add `data-testid="crud-item"`.
Same counts for the delete button with `data-testid="delete-crud"`.

Thoughts?

Reply:
I agree with your suggestions.

Reply:
@timdeschryver Is the CRUD example a demonstration of best practices in Angular? Are we also adding these tests to show as a good practice to follow ?

Reply:
Yea, it's to give an idea on how certain things are done with Angular. 
We do try to follow the best practices while doing so.

The same counts for these tests 😄